presto-cli/src/main/java/com/facebook/presto/cli/ErrorMessages.java

```diff
@@ -39,4 +72,80 @@ private static String getStackTraceString(Throwable throwable)
         throwable.printStackTrace(new PrintWriter(errors));
         return errors.toString();
     }
+
+    private static void technicalDetailsRuntimeExceptionErrorMessage(StringBuilder builder, Throwable throwable, ClientSession session)
+    {
+        builder.append(TECHNICAL_DETAILS_HEADER);
+        builder.append(ERROR_MESSAGE_INTRO);
+        builder.append(throwable.getMessage() + "\n\n");
+        builder.append(SESSION_INTRO);
+        builder.append(session + "\n\n");
+        builder.append(STACKTRACE_INTRO);
+        builder.append(getStackTraceString(throwable));
+        builder.append(TECHNICAL_DETAILS_END);
+    }
+
+    private static String standardQueryErrorMessage(StatementClient client)
+    {
+        StringBuilder builder = new StringBuilder();
+        QueryError error = extractQueryError(client);
+
+        builder.append(String.format("Query %s failed: %s%n", client.finalResults().getId(), error.getMessage()));
+        if (client.isDebug() && (error.getFailureInfo() != null)) {
+            StringWriter errors = new StringWriter();
+            error.getFailureInfo().toException().printStackTrace(new PrintWriter(errors));
+            builder.append(errors.toString());
+        }
+        if (error.getErrorLocation() != null) {
+            errorLocationMessage(builder, client.getQuery(), error.getErrorLocation());
+        }
+
+        return builder.toString();
+    }
+
+    private static void errorLocationMessage(StringBuilder builder, String query, ErrorLocation location)
+    {
+        List<String> lines = ImmutableList.copyOf(Splitter.on('\n').split(query).iterator());
+
+        String errorLine = lines.get(location.getLineNumber() - 1);
+        String good = errorLine.substring(0, location.getColumnNumber() - 1);
+        String bad = errorLine.substring(location.getColumnNumber() - 1);
+
+        if ((location.getLineNumber() == lines.size()) && bad.trim().isEmpty()) {
+            bad = " <EOF>";
+        }
+
+        if (REAL_TERMINAL) {
```

Comment:
This class shouldn't depend on ConsolePrinter, and shouldn't be trying to determine whether it's being used to output to a terminal in this way (e.g., it's possible that the console is outputting to the terminal, but this class is being invoked to render an error that will go to a log file).

A better way would be for this method to take an argument indicating whether it should output ANSI escape codes or not.